lua/pregeneration.lua

```diff
@@ -35,8 +35,9 @@ local function push_new_spawn()
                 rspawn:debug("Generated "..minetest.pos_to_string(newpos))
                 set_pgen(len_pgen()+1, newpos )
             else
-                rspawn:debug("Failed to generate new spawn point to push")
-                minetest.chat_send_player(rspawn.admin, "Failed to generate new spawn")
+                rspawn:debug("Failed to generate new spawn point to push, using predefined generated")
+                set_pgen(len_pgen()+1, random_pos )
+                minetest.chat_send_player(rspawn.admin, "Failed to generate new spawn, using predefined generated")
```

Comment:
⚠️ Are you sure this is what you want? `random_pos` could be anywhere - it could be in the middle of the stone core of a mountain, or out in the open sea....!

If you want to go this way, then I would advise creating a new setting `force_random_pregen_fallback` so that server admins can decide whether to use this feature...

Reply:
> warning Are you sure this is what you want? `random_pos` could be anywhere - it could be in the middle of the stone core of a mountain, or out in the open sea....!

is safe! cos there a already setting names "spawn_anywhere" .. if that is true.. that you said is happened (anywhere!) if false.. just set to zero point of fixed one defined in the minetest.conf

> 
> If you want to go this way, then I would advise creating a new setting `force_random_pregen_fallback` so that server admins can decide whether to use this feature...

is not necesary due the explanation above! i will made the changes requested by you and test again..

Reply:
umm.. umm.. ok ok.. but there's no worryes.. so currently then if setting is on false, then using my code will spawn in 0,0,0 or fixed defined in minetest.conf, if true.. will spawn anywhere inside limits of rspawn max and min..